### PR TITLE
Update for release 0.2

### PR DESCRIPTION
**Pull request by dave-fernandes**
The code to download the CIFAR images does not run on macOS, but the example trains if you download the data manually. A fix for downloading can probably wait until Dataset functionality is complete.

### CODE REVIEW

File: CIFAR/main.swift [hunk 1]
```diff
@@ -6,8 +6,9 @@ let (trainingDataset, testDataset) = loadCIFAR10()
 let trainingBatches = trainingDataset.batched(Int64(batchSize))
 let testBatches = testDataset.batched(Int64(batchSize))
 
-let learningPhaseIndicator = LearningPhaseIndicator()
-var model = CIFARModel() // learningPhaseIndicator: LearningPhaseIndicator)
+let trainContext = Context(learningPhase: LearningPhase.training)
```
Comment (rxwei) [inline CIFAR/main.swift]:
```suggestion
let trainContext = Context(learningPhase: .training)
```

File: CIFAR/main.swift [hunk 2]
```diff
@@ -6,8 +6,9 @@ let (trainingDataset, testDataset) = loadCIFAR10()
...
+let testContext = Context(learningPhase: LearningPhase.inference)
```
Comment (rxwei) [inline CIFAR/main.swift]: You can change call sites that use this to calling `inferring(from:)`.

File: CIFAR/models.swift [hunk 3]
```diff
@@ -46,49 +46,44 @@ public struct KerasModel : Layer {
     var dense1: Dense<Float>
     var dropout3: Dropout<Float>
     var dense2: Dense<Float>
-    public init(learningPhaseIndicator: LearningPhaseIndicator) {
+    public init(learningPhase: LearningPhase) {
```
Comment (rxwei) [inline CIFAR/models.swift]: You can remove this parameter completely.

```suggestion
    public init() {
```

File: CIFAR/models.swift [hunk 4]
```diff
@@ -46,49 +46,44 @@ public struct KerasModel : Layer {
...
+        tmp = dense2.applied(to: tmp, in: context)
         return tmp
     }
 }
 
 public typealias CIFARModel = PyTorchModel
 
 @differentiable(wrt: model)
-public func loss(model: CIFARModel, images: Tensor<Float>, labels: Tensor<Int32>)
+public func loss(model: CIFARModel, images: Tensor<Float>, labels: Tensor<Int32>, in context: Context)
```
Comment (rxwei) [inline CIFAR/models.swift]: Could you help inline this function body into the call site? I don't think this function should be defined as a standalone function. Perhaps it was defined this way to workaround some bugs.

File: CIFAR/main.swift (hunk 2 above)
Comment (dave-fernandes) [reply CIFAR/main.swift]: We would have to create a `CIFARModel.inferring(from:)` method to call the `Layer.inferring(from:)` methods. This seems to make more work than simply using `applied(to:in:)` everywhere. And `CIFARModel.inferring(from:)` would have to create a Context to call its `applied(to:in:)` or else recreate the whole model again using `Layer.inferring(from:)` calls.

Happy to make the change if this is the intent, but `inferring(from:)` seems more trouble than it is worth to me.

File: CIFAR/models.swift (hunk 4 above)
Comment (dave-fernandes) [reply CIFAR/models.swift]: It is actually used in two places -- training and test batch evaluation. I could move it to the main.swift file though, if that seems better.

File: CIFAR/main.swift (hunk 2 above)
Comment (dave-fernandes) [reply CIFAR/main.swift]: See comment below on loss function. That is what the Context is needed for.

File: CIFAR/models.swift (hunk 4 above)
Comment (rxwei) [reply CIFAR/models.swift]: Okay, I'm going to merge as is for now!

**Pull request merged**